src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -116,14 +130,178 @@ namespace OpenLoco::Vehicles
+        {
+            Vehicle train(component.head);
+            if (_vehicleUpdate_var_1136114 & (1U << 15))
+            {
+                if (train.veh1->routingHandle == component.routingHandle)
+                {
+                    _vehicleUpdate_var_1136114 |= (1U << 3);
+                    return false;
+                }
+            }
+            World::Pos3 pos(component.tileX, component.tileY, component.tileBaseZ * World::kSmallZStep);
+
+            auto [nextPos, nextRot] = World::Track::getTrackConnectionEnd(pos, component.trackAndDirection.track._data);
+            const auto tc = World::Track::getTrackConnections(nextPos, nextRot, component.owner, component.trackType, train.head->var_53, 0);
+            if (tc.hasLevelCrossing)
+            {
+                _vehicleUpdate_var_1136114 |= (1U << 4);
+            }
+            bool routingFound = false;
+            for (auto& connection : tc.connections)
+            {
+                if ((connection & 0x1FF) == (routing & 0x1FF))
+                {
+                    routingFound = true;
+                    break;
+                }
+            }
+            if (!routingFound)
+            {
+                _vehicleUpdate_var_1136114 |= (1U << 1);
+                return false;
+            }
+            component.routingHandle = newRoutingHandle;
+            const auto oldTaD = component.trackAndDirection.track._data;
+            component.trackAndDirection.track._data = routing & 0x1FF;
+            pos += World::TrackData::getUnkTrack(oldTaD).pos;
+            component.tileX = pos.x;
+            component.tileY = pos.y;
+            component.tileBaseZ = pos.z / World::kSmallZStep;
+            return true;
+        }
+
+        return false;
+    }
+
+    constexpr uint8_t getMovementNibble(const World::Pos3& pos1, const World::Pos3& pos2)
+    {
+        uint8_t nibble = 0;
+        if (pos1.x != pos2.x)
+        {
+            nibble |= (1U << 0);
+        }
+        if (pos1.y != pos2.y)
+        {
+            nibble |= (1U << 1);
+        }
+        if (pos1.z != pos2.z)
+        {
+            nibble |= (1U << 2);
+        }
+        return nibble;
+    }
+
+    // 0x00500120
+    constexpr std::array<uint32_t, 8> movementNibbleToDistance = {
+        0,
+        0x220C,
+        0x220C,
+        0x3027,
+        0x199A,
+        0x2A99,
+        0x2A99,
+        0x3689,
+    };
+
+    // 0x004B1876
+    static std::optional<EntityId> checkForCollisions(VehicleCommon& component, World::Pos3& loc)
+    {
+        registers regs{};
+        regs.esi = X86Pointer(&component);
+        regs.ax = loc.x;
+        regs.cx = loc.y;
+        regs.dx = loc.z;
+        auto res = call(0x004B1876, regs) & X86_FLAG_CARRY;
+        if (res)
+        {
+            return static_cast<EntityId>(regs.bp);
+        }
+        return std::nullopt;
+    }
+
+    static uint32_t updateTrackMotionT(VehicleCommon& component, uint32_t distance)
```

Comment:
Function should take an int32_t and return an int32_t

Reply:
Why the `T` suffix in `updateTrackMotionT`, btw? Leftover from when it was a templated function?

Reply:
Ah yeah i need to change that